src/classes/QueryArgumentFormatter.cls

```diff
@@ -6,42 +6,43 @@ public without sharing class QueryArgumentFormatter {
 
     private Object argumentValue;
 
-    public QueryArgumentFormatter(Object argumentValue) {
+    private QueryArgumentFormatter(Object argumentValue) {
```

Comment:
I would just change all of the methods to be static methods & completely eliminate the constructor